src/Controller/CRUDController.php

```diff
@@ -306,6 +306,19 @@ public function editAction($id = null)
             throw $this->createNotFoundException(sprintf('unable to find the object with id: %s', $id));
         }
 
+        if ($parentAdmin = $this->admin->getParent()) {
+            $parentId = $request->get($parentAdmin->getIdParameter());
+            $getterMethod = 'get'. ucfirst($this->admin->getParentAssociationMapping());
+        }
+
+        if ($parentAdmin && $parentAdmin->getObject($parentId) !== $existingObject->$getterMethod()) {
```

Comment:
shouldnt this if be inside the other if?

Reply:
It can be, I just wanted to avoid nesting ifs, should I move it?

Reply:
Lets figure out if this is the correct way to check this first.

Reply:
The only thing that concerns me is what if there is no getter? But then the only way to check it would be with reflection? cc @sonata-project/contributors any ideas?

Reply:
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Admin/AbstractAdmin.php#L2895

Here we obtain the value for the parent, through the relation, if I am reading this code correctly

Reply:
Well I can do it like that or just use reflection and finish this